### PR TITLE
feat: refactor `get_kes_period` to create isolated new queries

### PR DESCRIPTION
**Pull request by falcucci**
## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

### CODE REVIEW

**Comment by falcucci** [comment]
@jpraynaud ready for review. this can be useful for future use cases 😉

File: mithril-common/src/chain_observer/pallas_observer.rs [hunk 1]
```diff
@@ -580,7 +603,7 @@ mod tests {
     }
 
     /// Sets up a mock server.
-    async fn setup_server(socket_path: PathBuf) -> tokio::task::JoinHandle<()> {
+    async fn setup_server(socket_path: PathBuf, intersections: u32) -> tokio::task::JoinHandle<()> {
```
Comment (jpraynaud) [inline mithril-common/src/chain_observer/pallas_observer.rs:606]: Can you add a small comment to explain the way `intersections` work?

**Review by jpraynaud — APPROVED** [review]
Hi @falcucci, LGTM :rocket: 

Thanks for the contribution! 
I just left a minor comment below.

File: mithril-common/src/chain_observer/pallas_observer.rs (hunk 1 above)
Comment (falcucci) [reply mithril-common/src/chain_observer/pallas_observer.rs:610]: ofc, done!

**Review by Alenar — APPROVED** [review]
LGTM :)

**Comment by falcucci** [comment]
@jpraynaud done :)

**Pull request merged**